__tests__/app.test.js

```diff
@@ -45,25 +37,143 @@ describe('/api', () => {
+                    created_at: expect.any(String)
+                    })
+                })
+                expect(comments).toBeSorted({ descending: false });
+            })
+    });
+    test('POST:201', () => {
+        const newComment = {
+            author: 'tickle122',
+            comment: 'did  you know that peppa pig is actually vegetarian?',
+            article_id: 1
+            }
+        return request(app)
+        .post('/api/articles/1/comments')
+        .send(newComment)
+        .expect(201)
+        .then((response) => {
+            expect(response.body.comment.author).toBe(newComment.author);
+        expect(response.body.comment.body).toBe(newComment.comment);
+        expect(response.body.comment.article_id).toBe(newComment.article_id);
+        })
+    });
+})
+
+
+describe('Error Handling', () => {
+    describe('404: Bad Path', () => {
+        test('GET:404 sends an appropriate status and error message when given an invalid url', () => {
+        return request(app)
+        .get('/api/articules')
+        .expect(404)
+        .then((response) => {
+            expect(response.body.msg).toBe('Invalid URL');
+            })
+        });
+        test('GET:404 sends an appropriate status and error message when given an invalid url', () => {
+            return request(app)
+            .get('/api/topices')
+            .expect(404)
+            .then((response) => {
+            expect(response.body.msg).toBe('Invalid URL');
+            });
+        });
+        test('GET:404 sends an appropriate status and error message when provided with a non-existent but valid id', () => {
+            return request(app)
+            .get('/api/articles/999')
+            .expect(404)
+            .then((response) => {
+            expect(response.body.msg).toBe('Article does not exist');
+            });
+        });
+    });
+    test('POST: 404 Username not found', () => {
+        const newComment = {
+            author: 'peppapig',
+            comment: 'winner winner chicken dinner',
+            article_id: 2
+            }
+        return request(app)
+        .post('/api/articles/2/comments')
+        .send(newComment)
+        .expect(404)
+        .then((response) => {
+            expect(response.body.msg).toBe('Username not found');
+        })
+    });
+    test('GET: 404 Article does not have any comments', () => {
+        return request(app)
+        .get('/api/articles/37/comments')
+        .expect(404)
+        .then((response) => {
+            expect(response.body.msg).toBe("Article does not have any comments")
+        })
+    })
+    describe('GET:400 Bad Request', () => {
+        test('GET:400 sends an appropriate status and error message when provided with a invalid id', () => {
+            return request(app)
+            .get('/api/articles/abc')
+            .expect(400)
+            .then((response) => {
+            expect(response.body.msg).toBe('Bad Request');
+            });
```

Comment:
I would like one more test here: 201 ignores any unnecessary properties provided on the post body